plugins/inputs/haproxy/haproxy_test.go

```diff
@@ -134,17 +134,19 @@ func TestHaproxyGeneratesMetricsUsingSocket(t *testing.T) {
 	var randomNumber int64
 	var sockets [5]net.Listener
 
-	_globmask := filepath.Join(os.TempDir(), "test-haproxy*.sock")
-	_badmask := filepath.Join(os.TempDir(), "test-fail-haproxy*.sock")
+	// The Maximum length of the socket path is 104/108 characters, path created with t.TempDir() is too long for some cases
+	// (it combines test name with subtest name and some random numbers in the path)
+	//nolint:usetesting // "os.TempDir() could be replaced by t.TempDir() in TestHaproxyGeneratesMetricsUsingSocket"
+	tempDir := os.TempDir()
+	_globmask := filepath.Join(tempDir, "test-haproxy*.sock")
+	_badmask := filepath.Join(tempDir, "test-fail-haproxy*.sock")
```

Comment:
I don't see a removal of `tempDir`...

Reply:
It's probably good that you don't see it, such a removal probably wouldn't be possible :)

![image](https://github.com/user-attachments/assets/bf45cf75-6273-4d8c-b8a8-015e5bda6484)

Reply:
Oh I see. But the files are also not removed, are they?

Reply:
These are just globs used by plugin.
Socket files created later are automatically removed after socker is closed.